hooks.go

```diff
@@ -0,0 +1,113 @@
+package easypost
+
+import (
+	"context"
+	"github.com/google/uuid"
+	"io"
+	"net/url"
+	"time"
+)
+
+// HookEvent is the base type for all hook events
+type HookEvent struct {
+}
+
+// HookEventSubscriber is the base type for all hook event subscribers.
+// ID needs to be unique in order to find and remove the subscriber
+type HookEventSubscriber struct {
+	ID string // need unique ID for removal
+}
+
+// RequestHookEvent is the data passed to a RequestHookEventSubscriberCallback function
+type RequestHookEvent struct {
+	HookEvent        // implements HookEvent
+	Method           string
+	Url              *url.URL
+	RequestBody      io.ReadCloser
+	Headers          map[string][]string
+	RequestTimestamp time.Time
+	Id               uuid.UUID
+}
+
+// RequestHookEventSubscriberCallback is the type of the callback function executed by an RequestHookEventSubscriber
+type RequestHookEventSubscriberCallback func(ctx context.Context, event RequestHookEvent) error
+
+// RequestHookEventSubscriber is a HookEventSubscriber that executes a RequestHookEventSubscriberCallback function when a RequestHookEvent is fired
+type RequestHookEventSubscriber struct {
+	HookEventSubscriber // implements HookEventSubscriber
+	Callback            RequestHookEventSubscriberCallback
+}
+
+// Execute executes the RequestHookEventSubscriberCallback function of the RequestHookEventSubscriber
+func (s RequestHookEventSubscriber) Execute(ctx context.Context, event RequestHookEvent) {
+	err := s.Callback(ctx, event)
+	if err != nil {
+		return
+	}
+}
```

Comment:
```suggestion
// Executes the RequestHookEventSubscriberCallback function of the RequestHookEventSubscriber
func (s RequestHookEventSubscriber) Execute(ctx context.Context, event RequestHookEvent) {
	err := s.Callback(ctx, event)
	if err != nil {
		return
	}
}
```

Can we spell out what `s` is in the parameter?